cluster/etcd_service_discovery.go

```diff
@@ -490,21 +491,21 @@ func (p *parallelGetter) waitAndGetResult() []*Server {
 	return *p.result
 }
 
-func (p *parallelGetter) addWork(serverType, serverID string) {
+func (p *parallelGetter) addWork(serverType, serverID string, payload []byte) {
 	p.wg.Add(1)
 	p.workChan <- parallelGetterWork{
 		serverType: serverType,
 		serverID:   serverID,
+		payload:    payload,
 	}
 }
 
 // SyncServers gets all servers from etcd
 func (sd *etcdServiceDiscovery) SyncServers() error {
-	keys, err := sd.cli.Get(
+	kvs, err := sd.cli.Get(
 		context.TODO(),
 		"servers/",
 		clientv3.WithPrefix(),
-		clientv3.WithKeysOnly(),
```

Comment:
we better not do this every sync time. it's ok to do in server initialization though

Reply:
fixed in new commit